astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -1049,6 +1049,24 @@ class Lambda(_base_nodes.FilterStmtsBaseNode, LocalsDictNodeNG):
     special_attributes = FunctionModel()
     """The names of special attributes that this function has."""
 
+    args: Arguments
+    """The arguments that the function takes."""
+
+    body: NodeNG
```

Comment:
This is a change in behaviour and I think an inconsistency with `FunctionDef` which derives from this. `Lambda` seems to have a single node as a body whereas `FunctionDef` has a `list`. We might need to remove their coupling...

Reply:
I don't think this changes any behaviour, but yes, the `body` fields are inconsistent. Decoupling sounds like a good idea to me.

Reply:
Well, `mypy` will complain when we update the `FunctionDef` nodes right? So this change can't really be merged without doing the decoupling as well.

Reply:
That's just the tool telling us what we already know, which is that there are inconsistencies among the types. I don't think there's anything to be gained from leaving `body` unannotated in the two classes, especially since the typechecker warnings can be disabled.

Reply:
I think it better to leave them unannotated for now. Incorrect type annotations serve no purpose.

Reply:
Which one are you saying is incorrect? AFAICT, `Lambda.body` really is a single node, and `FunctionDef.body` really is a list.

Reply:
Well, `Mypy` won't be able to understand `FunctionDef.body` or at least warn us about it. The typing makes no sense with both a single node and a list.

We should focus on the refactor first before changing these two I think.

Reply:
Nothing is being changed though. Those fields already have the values that they have, namely a single node for `Lambda.body` and a list of nodes for `FunctionDef.body`. That is is current situation on the ground, and these annotations reflect that.

Mypy doesn't like the inheritance, hence the `type: ignore[assignment]` comments. But other than that it can figure out what's going on. If something is changed to give `FunctionDef.body` something other than a list, Mypy will flag it, and likewise for giving `Lambda.body` a list. If anything, Mypy can't understand things as they are now, since it believes that the two `body` fields have the same type, which they don't.